### PR TITLE
Add nopgo option for throughput runs

### PR DESCRIPTION
**Pull request by adiaaida**

### CODE REVIEW

File: perf.groovy [hunk 1]
```diff
@@ -184,82 +184,103 @@ def static getOSGroup(def os) {
...
+                                //we have to do it all as one statement because cmd is called each time and we lose the set environment variable
+                                batchFile("if \"%GIT_BRANCH:~0,7%\" == \"origin/\" (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH:origin/=%\") else (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH%\")\n" +
+                                "set \"BENCHVIEWNAME=${benchViewName}\"\n" +
+                                "set \"BENCHVIEWNAME=%BENCHVIEWNAME:\"=%\"\n" +
+                                "py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"${benchViewName}\" --user \"dotnet-bot@microsoft.com\"\n" +
+                                "py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\build.py\" git --branch %GIT_BRANCH_WITHOUT_ORIGIN% --type ${runType}")
+                                batchFile("py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\machinedata.py\"")
+                                batchFile("set __TestIntermediateDir=int&&build.cmd ${configuration} ${architecture}${pgo_build} skiptests")
```
Comment (dpodder) [inline perf.groovy]: For the sake of keeping folder paths shorter... could we use `${pgo_test}` here intead of `${pgo_build}`?

File: tests/scripts/run-throughput-perf.py [hunk 2]
```diff
@@ -200,7 +202,7 @@ def validate_arg(arg, check):
         benchview_path = os.path.normpath(benchview_path)
         validate_arg(benchview_path, lambda item: os.path.isdir(benchview_path))
 
-    args = (arch, operating_system, os_group, build_type, run_type, clr_root, assembly_root, benchview_path, iterations, opt_level, jit_name)
+    args = (arch, operating_system, os_group, build_type, run_type, clr_root, assembly_root, benchview_path, iterations, opt_level, jit_name, no_pgo)
 
     # Log configuration
     log('Configuration:')
```
Comment (dpodder) [inline tests/scripts/run-throughput-perf.py]: I think we should include whether PGO is on/off in the log here

File: perf.groovy (hunk 1 above)
Comment (dpodder) [reply perf.groovy]: Never mind... this isn't used in a dir path name. Please disregard.

File: tests/scripts/run-throughput-perf.py (hunk 2 above)
Comment (adiaaida) [reply tests/scripts/run-throughput-perf.py]: Sure thing. Good catch!

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet/rap-team @dotnet/eng-first does anyone know what's going on wiht the perf smoke tests? I'm seeing this error:

> 17:06:34 Using context: Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness
> 17:06:34 Building remotely on win2012-2b0a80 (win2012-20170809-elevated) in workspace D:\j\workspace\perf_perflab_---89ec9f56
> 17:06:34 [WS-CLEANUP] Deleting project workspace...
> 17:06:34 Selected Git installation does not exist. Using Default
> 17:06:34 Cloning the remote Git repository
> 17:06:34 Cloning repository https://github.com/dotnet/coreclr.git
> 17:06:34  > git init D:\j\workspace\perf_perflab_---89ec9f56 # timeout=10
> 17:06:34 Fetching upstream changes from https://github.com/dotnet/coreclr.git
> 17:06:34  > git --version # timeout=10
> 17:06:34  > git fetch --tags --progress https://github.com/dotnet/coreclr.git +refs/heads/*:refs/remotes/origin/* # timeout=90
> 17:06:35 
> ERROR: Error cloning remote repo 'origin'
> 17:06:35 hudson.plugins.git.GitException: Command "git fetch --tags --progress https://github.com/dotnet/coreclr.git +refs/heads/*:refs/remotes/origin/*" returned status code 128:
> 17:06:35 stdout: 
> 17:06:35 stderr: fatal: unable to access 'https://github.com/dotnet/coreclr.git/': error setting certificate verify locations:
> 17:06:35   CAfile: C:/Program Files/Git/mingw64/libexec/ssl/certs/ca-bundle.crt
> 17:06:35   CApath: none
> 17:06:35 
> 17:06:35 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1745)
> 17:06:35 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1489)
> 17:06:35 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$300(CliGitAPIImpl.java:64)
> 17:06:35 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:315)
> 17:06:35 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$2.execute(CliGitAPIImpl.java:512)
> 17:06:35 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:152)
> 17:06:35 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:145)
> 17:06:35 	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
> 17:06:35 	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
> 17:06:35 	at hudson.remoting.Request$2.run(Request.java:332)
> 17:06:35 	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
> 17:06:35 	at java.util.concurrent.FutureTask.run(Unknown Source)
> 17:06:35 	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
> 17:06:35 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
> 17:06:35 	at hudson.remoting.Engine$1$1.run(Engine.java:90)
> 17:06:35 	at java.lang.Thread.run(Unknown Source)
> 17:06:35 	at ......remote call to win2012-2b0a80(Native Method)
> 17:06:35 	at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1433)
> 17:06:35 	at hudson.remoting.UserResponse.retrieve(UserRequest.java:253)
> 17:06:35 	at hudson.remoting.Channel.call(Channel.java:797)
> 17:06:35 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:145)
> 17:06:35 	at sun.reflect.GeneratedMethodAccessor400.invoke(Unknown Source)
> 17:06:35 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 17:06:35 	at java.lang.reflect.Method.invoke(Method.java:498)
> 17:06:35 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:131)
> 17:06:35 	at com.sun.proxy.$Proxy164.execute(Unknown Source)
> 17:06:35 	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1042)
> 17:06:35 	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1082)
> 17:06:35 	at hudson.scm.SCM.checkout(SCM.java:495)
> 17:06:35 	at hudson.model.AbstractProject.checkout(AbstractProject.java:1278)
> 17:06:35 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:604)
> 17:06:35 	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
> 17:06:35 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:529)
> 17:06:35 	at hudson.model.Run.execute(Run.java:1720)
> 17:06:35 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
> 17:06:35 	at hudson.model.ResourceController.execute(ResourceController.java:98)
> 17:06:35 	at hudson.model.Executor.run(Executor.java:404)
> 17:06:35 ERROR: null

**Comment by mmitche** [comment]
@adiaaida I just mucked with that VM image.  Investgating

**Comment by mmitche** [comment]
@adiaaida Reverted to previous version.  The git installation was obviously corrupted when I upgraded it

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by adiaaida** [comment]
@mmitche Thanks!

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Pull request merged**